app/controllers/Application.scala

```diff
@@ -4,14 +4,35 @@ import play.api._
 import play.api.mvc._
 import services._
 
+import scala.concurrent.Future
+import scala.concurrent.ExecutionContext.Implicits.global
+
 object Application extends Controller {
 
   def index = Action {
     Ok(views.html.index("Your new application is ready."))
   }
 
-  def search(q: Option[String]) = Action {
+  def search(q: Option[String]) = Action.async {
     val results: Option[SearchResults] = q.map(SearchService.get)
-    Ok(views.html.search(q, results))
+
+    val seq : Option[Seq[SearchResultEntry]] = results.map{_.entries}
+
+    val withCodeSnippets = seq.map {
+      _.foldLeft(Future(Seq.empty[(String, SearchResultEntry)])){ (future, entry) =>
```

Comment:
Is this equivalent to the below?
```
Future.sequence(_.map(entry => SnippetFetcher.getSnippetCode(entry.repo, entry.path, entry.line)))
```
If yes I think that it is much easier to understand.